tests/org.eclipse.ui.tests/Eclipse UI Tests/org/eclipse/ui/tests/api/WorkbenchPluginTest.java

```diff
@@ -0,0 +1,85 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Vector Informatik GmbH and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ *******************************************************************************/
+package org.eclipse.ui.tests.api;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.Assume.assumeFalse;
+
+import java.util.concurrent.atomic.AtomicReference;
+
+import org.eclipse.core.runtime.Platform.OS;
+import org.eclipse.core.tests.harness.TestBarrier2;
+import org.eclipse.jface.resource.ImageDescriptor;
+import org.eclipse.jface.resource.ImageRegistry;
+import org.eclipse.swt.graphics.Image;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.ui.PlatformUI;
+import org.eclipse.ui.tests.TestPlugin;
+import org.junit.Test;
+
+public class WorkbenchPluginTest {
+
+	@Test
+	public void testGetImageRegistryFromAdditionalDisplay() {
+		assumeFalse("multiple displays are not allowed on Linux", OS.isLinux());
+		assumeFalse("multiple displays are not allowed on MaxOS", OS.isMac());
+		String dummyFilename = "test"; //$NON-NLS-1$
+		Display displayInOtherThread = initializeDisplayInOtherThread();
+		try {
+			displayInOtherThread.syncExec(() -> getFileIcon(dummyFilename));
+			final Image iconFromMainThread = getFileIcon(dummyFilename);
+			disposeDisplay(displayInOtherThread);
+			assertThat("icon retrieved in main thread has been disposed through other display",
+					iconFromMainThread.isDisposed(), is(false));
+		} finally {
+			disposeDisplay(displayInOtherThread);
+		}
+	}
+
+	private Image getFileIcon(String filename) {
+		ImageRegistry registry = TestPlugin.getDefault().getImageRegistry();
+		Image fileIcon = registry.get(filename);
+		if (fileIcon != null) {
+			return fileIcon;
+		}
+		ImageDescriptor imageDescriptor = PlatformUI.getWorkbench().getEditorRegistry().getImageDescriptor(filename);
+		if (imageDescriptor != null) {
+			registry.put(filename, imageDescriptor);
+		}
+		return registry.get(filename);
+	}
```

Comment:
This one is pretty similar to `org.eclipse.ui.ISharedImages.getImage(String)` and it currently only works because the `EditorRegistry` ends up providing a default image (see `org.eclipse.ui.internal.registry.EditorRegistry.getDefaultImage()`). You could directly call `org.eclipse.ui.ISharedImages.getImage(String)` and use the descriptor of the default image:

```suggestion
	private Image getFileIcon() {
		return PlatformUI.getWorkbench().getSharedImages().getImage(ISharedImages.IMG_OBJ_FILE);
	}
```

Reply:
I think I do not really understand the suggestion. The provided test is a regression test for the issue fixed with the PR. When I change the test code as proposed, the test will succeed, but it will not be a regression test for the issue anymore (it then already succceeds with the existing code).

Reply:
You're right, you're testing `AbstractUIPlugin` through `TestPlugin`. In that case I suggest you make the test independent from `TestPlugin` in order to clarify this intent, you could use: `ImageRegistry registry = new AbstractUIPlugin() {}.getImageRegistry();`

Reply:
Yes, that's a good idea. In particular because it is important that the tested plug-in's image registry is initialized from another thread, which may not be guaranteed for the `TestPlugin` in case some other test already accessed it. I have adapter the test in e3af6f0 with one difference to your proposal: the plugin is kept as an instance variable as otherwise every call to an image access will be done on a new plug-in instance (which would then again lead to the test not being a regression test for the fixed issue).